http-client.ts

```diff
@@ -266,16 +277,19 @@ export class HttpClient implements Server.IHttpClient {
 	 */
 	private async useProxySettings(proxySettings: IProxySettings, proxyCache: IProxyCache, options: any, headers: any, requestProto: string): Promise<void> {
 		if (proxySettings || proxyCache) {
-			options.path = requestProto + "://" + options.host + options.path;
-			headers.Host = options.host;
-			options.host = (proxySettings && proxySettings.hostname) || proxyCache.PROXY_HOSTNAME;
-			options.port = (proxySettings && proxySettings.port) || proxyCache.PROXY_PORT;
-
+			const proto = (proxySettings && proxySettings.protocol) || proxyCache.PROXY_PROTOCOL || "http:";
+			const host = (proxySettings && proxySettings.hostname) || proxyCache.PROXY_HOSTNAME;
+			const port = (proxySettings && proxySettings.port) || proxyCache.PROXY_PORT;
+			let credentialsPart = "";
 			const proxyCredentials = await this.$proxyService.getCredentials();
 			if (proxyCredentials && proxyCredentials.username && proxyCredentials.password) {
-				headers["Proxy-Authorization"] = "Basic " + new Buffer(`${proxyCredentials.username}:${proxyCredentials.password}`).toString('base64');
+				credentialsPart = `${proxyCredentials.username}:${proxyCredentials.password}@`;
 			}
 
+			// Note that proto ends with :
+			options.proxy = `${proto}//${credentialsPart}${host}:${port}`;
+			options.rejectUnauthorized = false;
```

Comment:
Why is this required? Maybe this should be an option in the proxy configuration.